pkg/kubectl/resource/result.go

```diff
@@ -160,14 +160,22 @@ func (r *Result) Object() (runtime.Object, error) {
 	if len(versions) == 1 {
 		version = versions.List()[0]
 	}
-	return &api.List{
+	raw := []runtime.RawExtension{}
+	for _, o := range objects {
+		raw = append(raw, runtime.RawExtension{Object: o})
+	}
+	return &v1.List{
```

Comment:
needs godoc update (and sweeping callers to catch casts/tests for api.List instead of v1.List)

> // Otherwise, the returned resources will be part of an
> // api.List. The ResourceVersion of the api.List will be set only if it is identical
> // api.List

Reply:
I checked for callers of the function and did a rough analysis of how each is using it.  Updated the PR description with each caller of the function, and how they use the result.